src/include/wiredtiger.in

```diff
@@ -3682,6 +3704,502 @@ struct __wt_extractor {
 	int (*terminate)(WT_EXTRACTOR *extractor, WT_SESSION *session);
 };
 
+#if !defined(SWIG)
+/*! WT_FILE_SYSTEM::open_file type: checkpoint */
+#define	WT_FILE_TYPE_CHECKPOINT	1
+/*! WT_FILE_SYSTEM::open_file type: read/write data object */
+#define	WT_FILE_TYPE_DATA	2
+/*! WT_FILE_SYSTEM::open_file type: directory */
+#define	WT_FILE_TYPE_DIRECTORY	3
+/*! WT_FILE_SYSTEM::open_file type: log file */
+#define	WT_FILE_TYPE_LOG	4
+/*! WT_FILE_SYSTEM::open_file type: regular file */
+#define	WT_FILE_TYPE_REGULAR	5
```

Comment:
This would be cleaner and more type-safe as an `enum` -- any reason not to do that?